addons/controllers/clusterbootstrap_controller.go

```diff
@@ -618,10 +618,13 @@ func (r *ClusterBootstrapReconciler) createOrPatchKappPackageInstall(clusterBoot
 		if err != nil {
 			return err
 		}
-		pkgi.Spec.Values = []kapppkgiv1alpha1.PackageInstallValues{
-			{SecretRef: &kapppkgiv1alpha1.PackageInstallValuesSecretRef{
-				Name: secret.Name},
-			},
+
+		if secret != nil {
```

Comment:
Thanks for looking into this. I think we should look at why the secret being returned from the prev step is nil and the bug is probably in that function

Reply:
Hi @vijaykatam , I found that I can reproduce this crash error when using custom `Clusterbootstrap` with `spec.kapp.valuesFrom.providerRef` points to a `KappControllerConfig`, which has no ownerReference and its name != Cluster name. This means this `KappControllerConfig` never got reconciled due to its corresponding Cluster not found.
https://github.com/vmware-tanzu/tanzu-framework/blob/a2cd9c3632002170c8bc7090c097363181bbb69f/addons/controllers/kapp-controller/kappcontrollerconfig_controller.go#L62-L80

So the data values secret of kapp package was never generated and the status of that KappControllerConfig was never updated. Yet this "unreconciled" KappControllerConfig is used to bootstrap a cluster, and the empty status was treated as no configurations for kapp package.

Based on my understanding, the "no configurations package" is allowed in the general idea, since some packages do not need extra configurations. But for the kapp package as specific, the configuration should always exist, right? 

So maybe we can add a verification to check if the returned secret is empty. If so, we should throw an error instead of continue bootstrapping the cluster. This error would block the cluster bootstrap, until the status of the KappControllerConfig shows the data values secret(no matter for timing reason or misconfiguration correction). What do you think?

Reply:
Thanks for digging through. I think we are not handling this specific case correctly. I looked at the other instance of usage of `createOrPatchPackageInstallSecret` and we are handling it by not setting packageinstall.Spec.Values. For consistency I am going to do the same for kapp-controller with a warning in the logs.

Reply:
I see. Great! Thanks @vijaykatam . I will update the details to the issue and close this draft pr, since you are already looking into this.